cdist/conf/type/__user/gencode-remote

```diff
@@ -130,13 +131,17 @@ if [ "$state" = "present" ]; then
           echo useradd "$@" "$name"
        fi
     fi
-else
+elif [ "$state" = "absent" ]; then
     if grep -q "^${name}:" "$__object/explorer/passwd"; then
         #user exists, but state != present, so delete it
         if [ -f "$__object/parameter/remove-home" ]; then
-            echo userdel -r "${name}"
+            printf "userdel -r %s >/dev/null 2>&1\n" "${name}"
```

Comment:
I'd suggest single quotes here, so the name arrives quoted on the target.

Reply:
I was hesitant about this one, as users can (should?) not contain whitespace -- but better safe that sorry here ...